lib/jsdom/virtual-console.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+
+var EventEmitter = require("events").EventEmitter;
+
+var vConsole = {};
+
+var emitters = {};
+var windows = [];
+
+// Returns an EventEmitter that emits on any console events
+vConsole.getVirtualConsole = function(window) {
+  var virtualConsole = new EventEmitter();
+
+  saveEmitterForWindow(virtualConsole, window);
+
+  virtualConsole.sendTo = function(anyConsole) {
+
+    if (!isValidConsole(anyConsole, window)) {
+      var msg = "sendTo was not passed a compliant console";
+      throw new Error(msg);
+    }
+
+    var vc = vConsole.getVirtualConsole(window);
+    Object.keys(anyConsole).forEach(function(method) {
+      if (typeof anyConsole[method] === "function") {
+        vc.on(method, anyConsole[method]);
```

Comment:
Node.js console methods are functions, i.e. don't require `this`-binding. But in general that's not a good assumption. So instead, `vc.on(method, function () { anyConsole[method].apply(anyConsole, arguments); })`